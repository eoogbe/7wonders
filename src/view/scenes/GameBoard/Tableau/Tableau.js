/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import './Tableau.css';
import TableauCard from './TableauCard';

const renderCard = (cardActions) => (card) =>(
  <TableauCard
    key={card.id}
    {...cardActions}
    {...card}
  />
);

const renderNonResourceCards = ({ cards, ...cardActions }) => {
  if (!cards.length) {
    return <></>;
  }

  return (
    <ul className="Tableau-cardList">
      {cards.map(renderCard(cardActions))}
    </ul>
  );
}

const Tableau = ({
  style, resourceCards, nonResourceCards, showTableauCallout,
}) => (
  <div className="Tableau" style={style || {}}>
    <ul className="Tableau-cardList">
      {resourceCards.map(renderCard({ showTableauCallout }))}
    </ul>
    {renderNonResourceCards({ cards: nonResourceCards, showTableauCallout })}
  </div>
);

export default Tableau;
