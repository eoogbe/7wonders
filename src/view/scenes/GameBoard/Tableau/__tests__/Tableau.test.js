/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Tableau from '../Tableau';

const RESOURCE_CARDS = [
  {
    id: 'PYRAMIDS_OF_GIZA_INITIAL',
    type: 'RAW_MATERIAL',
    effects: ['Stone'],
  },
];

const NON_RESOURCE_CARDS = [
  {
    id: 'STOCKADE',
    type: 'MILITARY',
    effects: ['1 shield'],
  },
  {
    id: 'TAVERN',
    type: 'TRADE',
    effects: ['5 coins'],
  },
];

jest.useFakeTimers();

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <Tableau
          resourceCards={RESOURCE_CARDS}
          nonResourceCards={NON_RESOURCE_CARDS}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with only resource cards', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <Tableau
          resourceCards={RESOURCE_CARDS}
          nonResourceCards={[]}
        />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('shows the tableau callout', () => {
  const showTableauCallout = jest.fn();
  const store = mockStore({});
  const tree = mount(
    <Provider store={store}>
      <Tableau
        resourceCards={RESOURCE_CARDS}
        nonResourceCards={NON_RESOURCE_CARDS}
        showTableauCallout={showTableauCallout}
      />
    </Provider>
  );
  tree.find('.TableauCard').first().simulate('mouseEnter');
  expect(showTableauCallout).toHaveBeenCalledWith('PYRAMIDS_OF_GIZA_INITIAL');
});
