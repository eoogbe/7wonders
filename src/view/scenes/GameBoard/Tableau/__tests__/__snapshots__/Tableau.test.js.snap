// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with only resource cards 1`] = `
<div
  className="Tableau"
  style={Object {}}
>
  <ul
    className="Tableau-cardList"
  >
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(141,110,99)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          Stone
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="Tableau"
  style={Object {}}
>
  <ul
    className="Tableau-cardList"
  >
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(141,110,99)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          Stone
        </li>
      </ul>
    </li>
  </ul>
  <ul
    className="Tableau-cardList"
  >
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(239,83,80)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          1 shield
        </li>
      </ul>
    </li>
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(255,235,59)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          5 coins
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
